src/main/resources/testchipip/csrc/SimDRAM.cc

```diff
@@ -25,6 +25,14 @@ extern "C" void *memory_init(
                              long long int mem_base
 			     )
 {
+    mem_size &= 0xFFFFFFFF;
+    word_size &= 0xFFFFFFFF;
+    line_size &= 0xFFFFFFFF;
+    id_bits &= 0xFFFFFFFF;
+    clock_hz &= 0xFFFFFFFF;
+    mem_base &= 0xFFFFFFFF;
```

Comment:
Maybe we should mask based on the ADDR_SIZE parameter (pass that into these functions and use it to mask at least mem_base)?

Reply:
Any thoughts on my mem_base might be sign extended in this function (the parameter in `SimDRAM.v` is a 40b `0x80000000` but when you print it's output here it is `0xFFFF8....`).